x-pack/plugins/lists/server/saved_objects/migrations.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { SavedObjectSanitizedDoc, SavedObjectUnsanitizedDoc } from 'kibana/server';
+
+import { ENDPOINT_LIST_ID, ENDPOINT_TRUSTED_APPS_LIST_ID } from '../../common/constants';
+import {
+  EntriesArray,
+  ExceptionListSoSchema,
+  NonEmptyNestedEntriesArray,
+  OsTypeArray,
+  entriesNested,
+  entry,
+} from '../../common/schemas';
+
+const entryType = t.union([entry, entriesNested]);
+type EntryType = t.TypeOf<typeof entryType>;
+
+const migrateEntry = (entryToMigrate: EntryType): EntryType => {
+  const newEntry = entryToMigrate;
+  if (entriesNested.is(entryToMigrate) && entriesNested.is(newEntry)) {
+    newEntry.entries = entryToMigrate.entries.map((nestedEntry) =>
+      migrateEntry(nestedEntry)
+    ) as NonEmptyNestedEntriesArray;
+  }
+  newEntry.field = entryToMigrate.field.replace('.text', '.caseless');
+  return newEntry;
```

Comment:
does making `newEntry` do anything here? it looks like a reference to the same object as `entryToMigrate`. The `if` conditions appear to be redundant as well.

Reply:
@marshallmain Thanks, will fix!

Reply:
@marshallmain Oh, this was actually to get around the `no-param-reassign` eslint rule. Leaving it as-is.